src/database/DataTypes/StocktakeItem.js

```diff
@@ -142,8 +142,8 @@ export class StocktakeItem extends Realm.Object {
    * Finds the mode option within this stocktake items batches
    * @return {string} The title of the reason with the highest frequency
    */
-  get mostUsedReason() {
-    if (!this.batches.length) return false;
+  get mostUsedReasonTitle() {
```

Comment:
Just wondering if it's worth using realm to filter batches without an option before iterating?

```
if (!this.batches.length) return '';
const batchesWithAReason = this.batches.filtered('option !== null')(???);
if (!batchesWithAReason) return '';
```

bit ugly but could be worth it?

Reply:
The following existing code will pretty quickly evaluate the same thing without hitting up realm with a filter query
```javascript
this.batches.forEach(batch => {
      const { option } = batch;
      if (!option) return;
...
```

Reply:
I'd put this in the premature optimisation bucket but I'm going to do some science anyway

Reply:
For a stocktakeItem with 30 batches, 15 with a reason, 15 without. 10000 runs averages:
(got surprising variance, but I'll just take worst run of 4 and not spend more time on this lol)

Without this tweak: 0.9065ms
With this tweak: 1.0453ms

30 batches, no diff no reasons:

Without this tweak: 0.6661ms
With this tweak: 0.1110ms

**Take away:**
- Unless we have a stocktake covering over say 1000 batches, probably no point optimising here. Unless something is noticeably slow, not worth investigating (I wasted an hour... maybe 2 now doing this, even if it was fun)

Hilariously, somehow, the OG optimisation slows things down? `if (!this.batches.length) return '';`

Without:
![Screen Shot 2019-05-29 at 12 45 05](https://user-images.githubusercontent.com/7684221/58521271-1d0d2880-8210-11e9-8938-f4ac746f18a5.png)
Without:
![Screen Shot 2019-05-29 at 12 46 11](https://user-images.githubusercontent.com/7684221/58521214-e2a38b80-820f-11e9-908f-04bee1dfb10e.png)

I suspect that might be the JS JIT compiler is having trouble optimising the function with our optimisation in there.

Anyway, I'm removing the old one and adding yours and moving on Jesus christ me

Reply:
I'd argue that it's probably worse adding this change:
- Places that don't use reasons and therefore don't render the column, it makes no difference
- Most stocktakeItems are going to have a few batches with at least one reason, so this will likely just slow down the stocktake as they fill it out
- Unless it's a freaking huge stocktake, it's negligible. And still nothing on the time wasted on unnecessary re-renders!
- If I spend a single keystroke more on this put me in the naughty corner to think about what I've done.

Oh god I couldn't stop:
On an item with 1 batch, one reason, 10000 run average:

Without: ~0.075ms
With: ~0.185ms

Yea it's ded let's stop pls god

Reply:
I wonder though if this should be calculated in the component cached and updated as reasons are added to help in the render cy.... hehehhee

Reply:
If the shallow compare saw that the object hasn't changed, shouldn't rerender/re-get the cell/value I think. Blah blah new table things